File: Solutions by Programming Task/Fibonacci sequence/Java/Iterative1.java
import java.util.Arrays;
import java.util.stream.IntStream;

public class Main {
    private static int Fib(int x) {
        var data = new int[]{0, 1};
        IntStream.range(0, x).forEach(__ -> {
            var sum = Arrays.stream(data).sum();
            data[0] = data[1];
            data[1] = sum;
        });
        return data[0];
    }

    public static void main(String[] args) {
        System.out.println(Fib(10));
    }
}

File: Solutions by Programming Task/Fibonacci sequence/Go/Iterative1.go
package main

import "fmt"

func Fib(x int) int {
	p, n := 0, 1
	for range make([]int, x) {
		p, n = n, p + n
	}
	return p
}

func main() {
	fmt.Println(Fib(10))
}
